ge/org.osate.ge/src/org/osate/ge/internal/ui/handlers/DeleteHandler.java

```diff
@@ -81,57 +144,225 @@ public Object execute(final ExecutionEvent event) throws ExecutionException {
 			throw new RuntimeException("Unexpected editor: " + activeEditor);
 		}
 
+		final AgeDiagramEditor ageEditor = (AgeDiagramEditor) activeEditor;
+
+		final ExtensionService extensionService = Objects.requireNonNull(
+				(ExtensionService) ageEditor.getAdapter(ExtensionService.class),
+				"Unable to retrieve extension service");
+		final ReferenceBuilderService refBuilder = Objects.requireNonNull(
+				(ReferenceBuilderService) ageEditor.getAdapter(ReferenceBuilderService.class),
+				"Unable to retrieve reference builder service");
+		final AadlModificationService aadlModificationService = Objects.requireNonNull(
+				(AadlModificationService) ageEditor.getAdapter(AadlModificationService.class),
+				"Unable to retrieve modification service");
+
 		// Get diagram and selected elements
 		final List<DiagramElement> selectedDiagramElements = AgeHandlerUtil.getSelectedDiagramElements();
-		if (selectedDiagramElements.size() == 0) {
-			throw new RuntimeException("No element selected");
+		if (!canExecute(selectedDiagramElements, extensionService)) {
+			throw new RuntimeException("canExecute() returned false");
 		}
 
-		final AgeDiagramEditor ageEditor = (AgeDiagramEditor) activeEditor;
+		if (!confirmDelete(selectedDiagramElements)) {
+			return null;
+		}
 
-		final GraphitiAgeDiagram graphitiAgeDiagram = Objects.requireNonNull(ageEditor.getGraphitiAgeDiagram(),
-				"Unable to retrieve graphiti diagram");
+		final boolean boIsContext = anyBoIsDiagramContext(selectedDiagramElements, ageEditor.getAgeDiagram(),
+				refBuilder);
 
-		final IFeatureProvider fp = Objects.requireNonNull(ageEditor.getDiagramTypeProvider().getFeatureProvider(),
-				"Unable to retrieve feature provider");
+		ageEditor.getActionExecutor().execute("Delete", ExecutionMode.NORMAL, () -> {
+			if (anyRequiresRawDelete(selectedDiagramElements)) {
+				if (selectedDiagramElements.size() != 1) {
+					throw new RuntimeException("Deleting multiple elements when using DeleteRaw is not supported");
+				}
 
-		final Map<DeleteContext, IDeleteFeature> deleteContextToFeatureMap = new LinkedHashMap<>();
-		for (final DiagramElement de : selectedDiagramElements) {
-			final PictogramElement pe = Objects.requireNonNull(graphitiAgeDiagram.getPictogramElement(de),
-					"Unable to retrieve pictogram element");
-			final DeleteContext deleteContext = new DeleteContext(pe);
-			final IDeleteFeature deleteFeature = fp.getDeleteFeature(deleteContext);
-
-			if (selectedDiagramElements.size() > 1) {
-				deleteContext
-				.setMultiDeleteInfo(new MultiDeleteInfo(false, false, selectedDiagramElements.size()));
+				deleteRaw(selectedDiagramElements.get(0), extensionService);
 			}
+			else if (anyIsInAnnex(selectedDiagramElements)) {
```

Comment:
else if is on the wrong line

Reply:
@RyanMcilnay should be fixed now. I reran the formatter.